modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastsViewModel.kt

```diff
@@ -300,9 +302,11 @@ class PodcastsViewModel
     suspend fun loadSuggestedFolders() {
         if (FeatureFlag.isEnabled(Feature.SUGGESTED_FOLDERS)) {
             _suggestedFoldersState.emit(SuggestedFoldersState.Loading)
-            val folders = suggestedFoldersManager.getSuggestedFolders().toFolders()
-            if (!folders.isEmpty()) {
-                _suggestedFoldersState.emit(SuggestedFoldersState.Loaded(folders))
+            suggestedFoldersManager.getSuggestedFolders().collect { state ->
```

Comment:
Does this need to be a flow? If it does, should the calling method use a `repeatOnLifecycle`?

Reply:
Yes and yes! I will update it and merge the PR